src/python/pose_format/bin/directory.py

```diff
@@ -1,45 +1,171 @@
 import argparse
-import os
-
+from pathlib import Path
 from pose_format.bin.pose_estimation import pose_video, parse_additional_config
+from typing import List
+import logging
 from tqdm import tqdm
 
+# Note: untested other than .mp4. Support for .webm may have issues: https://github.com/sign-language-processing/pose/pull/126
+_SUPPORTED_VIDEO_FORMATS = [".mp4", ".mov", ".avi", ".mkv", ".flv", ".wmv", ".webm"]
+
+
+def find_videos_with_missing_pose_files(
+    directory: Path,
+    video_suffixes: List[str] = None,
+    recursive: bool = False,
+    keep_video_suffixes: bool = False,
+) -> List[Path]:
+    """
+    Finds videos with missing .pose files.
+
+    Parameters
+    ----------
+    directory: Path,
+        Directory to search for videos in.
+    video_suffixes:  List[str], optional
+        Suffixes to look for, e.g. [".mp4", ".webm"]. If None, will use _SUPPORTED_VIDEO_FORMATS
+    recursive: bool, optional
+        Whether to look for video files recursively, or just the top-level. Defaults to false.
+    keep_video_suffixes: bool, optional
+        If true, when checking will append .pose suffix (e.g. foo.mp4->foo.mp4.pose, foo.webm->foo.webm.pose), 
+        If false, will replace it (foo.mp4 becomes foo.pose, and foo.webm ALSO becomes foo.pose). 
+        Default is false, which can cause name collisions.
+
+    Returns
+    -------
+    List[Path]
+        List of video paths without corresponding .pose files.
+    """
+
+    # Prevents the common gotcha with mutable default arg lists: 
+    # https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments
+    if video_suffixes is None:
+        video_suffixes = _SUPPORTED_VIDEO_FORMATS
+
+    glob_method = getattr(directory, "rglob" if recursive else "glob")
+    all_files = list(glob_method(f"*"))
+    video_files = [path for path in all_files if path.suffix in video_suffixes]
+    pose_files = [path for path in all_files if path.suffix == ".pose"]
 
-def removesuffix(text: str, suffix: str):
-	if text.endswith(suffix):
-		return text[:-len(suffix)]
-	else:
-		return text
+    videos_with_missing_pose_files = []
 
+    for vid_path in video_files:
+        if (
+            not get_corresponding_pose_path(
```

Comment:
the if statement is ugly. use a variable `corresponding_pose_path = get....` and then check